src/test/java/StringManipulationTest.java

```diff
@@ -89,23 +101,59 @@ public void testGeSubStrings1() {
 
     @Test
     public void testGeSubStrings2() {
-        fail("Not yet implemented");
-    }
-    @Test
-    public void testGeSubStrings3() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("This is a rather long, overblown, drawn-out, random sentence just used for test purposes.");
+        String [] sStings = manipulatedstring.getSubStrings(1, 14);
+
+        assertEquals(sStings[0], "This" );
+        assertEquals(sStings[1], "is" );
+        assertEquals(sStings[2], "a" );
+        assertEquals(sStings[3], "rather" );
+        assertEquals(sStings[4], "long" );
+        assertEquals(sStings[5], "overblown" );
+        assertEquals(sStings[6], "drawn-out" );
+        assertEquals(sStings[7], "random" );
+        assertEquals(sStings[8], "sentence" );
+        assertEquals(sStings[9], "just" );
+        assertEquals(sStings[10], "used" );
+        assertEquals(sStings[11], "for" );
+        assertEquals(sStings[12], "test" );
+        assertEquals(sStings[13], "purposes" );
+    }
+    @Test
+    public void testGeSubStrings3(){
+        manipulatedstring.setString("Hello World");
+        assertThrows(java.lang.IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(0, 1));
     }
     @Test
     public void testGeSubStrings4() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello World");
+        assertThrows(java.lang.IndexOutOfBoundsException.class, () -> manipulatedstring.getSubStrings(1, 4));
     }
+
+
     @Test
     public void testGeSubStrings5() {
-        fail("Not yet implemented");
+        manipulatedstring.setString("Hello, World");
+        String [] sStings = manipulatedstring.getSubStrings(1, 2);
+        assertEquals(sStings[0], "Hello" );
```

Comment:
Good work checking the shorter strings. Making sure to test all kinds of cases is important.